app/src/main/java/com/github/orkest/View/FollowListActivity.kt

```diff
@@ -0,0 +1,220 @@
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.material3.IconButton
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.orkest.Model.FireStoreDatabaseAPI
+import com.github.orkest.Model.Profile
+import com.github.orkest.Model.User
+import com.github.orkest.View.profile.ProfileActivity
+import com.github.orkest.View.sharing.ui.theme.OrkestTheme
+import com.github.orkest.ViewModel.FollowListViewModel
+import com.github.orkest.R
+
+//Here for preview purposes. Will be deleted.
+val user = User(username = "Philippe", profile = Profile(profilePictureId = R.drawable.blank_profile_pic))
+
+private val paddingValue = 5.dp
+private val titleFontSize = 28.sp
+private val textFontSize = 18.sp
+private val smallSeparator = 10.dp
+private val largeSeparator = 20.dp
+private val itemSize = 50.dp
+private val roundedCornerValue = 20.dp
+
+/**
+ * This activity is launched when a user wants to display the list of followers or followings of another user (or himself)
+ * Two extras are sent with the intent to launch the activity
+ * username : String = The username of the user to which we want to access the follow lists
+ * isFollowers : Boolean = Whether it is the followers list we want to display or the followings list
+ */
+class FollowListActivity: ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            val username = intent.getStringExtra("username").toString()
+            val isFollowers = intent.getBooleanExtra("isFollowers", false)
+            OrkestTheme {
+                FollowList(viewModel = FollowListViewModel(username, isFollowers))
+            }
+        }
+    }
+}
+
+
+@Composable
+fun FollowList(viewModel: FollowListViewModel){
+    Column {
+        Row(Modifier
+            .fillMaxWidth()
+            .background(Color.Yellow)
+        ) {
+            val context = LocalContext.current
+            IconButton(onClick = {
+                // Gets back to the profile page of the user

```

Comment:
Maybe it could be nice to be able to go back to the main activity, or maybe it is already possible and did not saw it

Reply:
When clicking on the "Back button", we get back to the user's profile so it is already implemented